### PR TITLE
fix(tag): allow use of onClick when onClose is not present

### PR DESCRIPTION
**Pull request by tw15egan**
Screen Shot 2020-05-06 at 10 58 29 AM

This change should allow `onClick` to bubble up if no `onClose` is present

#### Changelog

**Changed**

- Checks for the presence of `onClick` and `onClose` and uses the one a user provides. 

#### Testing / Reviewing

`OnClose` should fire in the default storybook environment. Comment out the `onClose` action in the tag filter props, and ensure the `onClick` action fires.

### CODE REVIEW

File: packages/react/src/components/Tag/Tag.js [hunk 1]
```diff
@@ -46,8 +47,13 @@ const Tag = ({
     [`${prefix}--tag--${type}`]: type,
   });
   const handleClose = event => {
-    event.stopPropagation();
-    onClose(event);
+    if (onClose) {
+      event.stopPropagation();
+      onClose(event);
+    }
+    if (onClick && !onClose) {
```
Comment (electrostaticfleece) [inline packages/react/src/components/Tag/Tag.js]: Wondering if `onClick` is necessary here and if the onClose condition check is sufficient? Should this not just bubble up to the div where the `onClick` handler was originally located and where the `...other` props are spread?

Comment (tw15egan) [reply packages/react/src/components/Tag/Tag.js]: Ah right, updated it now 👍

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Though I wonder why `event.stopPropagation()` is called there in the first place, but this fix itself is fine. Thanks @tw15egan!

**Comment by emyarod** [comment]
IIRC the `stopPropagation` separates the `onClick` and `onClose` events, although if we want `onClick` to also fire when the close button is clicked, we can remove that

**Review by emyarod — APPROVED** [review]
looks good to me

**Pull request merged**